### PR TITLE
Beautify Progress for Deploy

### PR DESCRIPTION
**Pull request by breeswish**
This is an initial implementation, including a framework, as well as changed the `deploy` command. Multiple progress bars in parallel is also supported.

- Currently TiDB Component download progress bar is not overridden, since it is printed by TiUP. TODO: Update TiUP to not displaying a progress bar when downloading components..

- When there are too many multi progress bars (number > screen height), the display may be buggy. Please only display a limited number of multi progress bars for now. For example, do not display a progress bar for each host, since the host may be 100+.

image

You can customize the text after the spinner for each "step" using `ctx.ev.PublishTaskProgress(s, "...")`. Currently it display `task.String()` by default.

Errors are displayed as:

image

### CODE REVIEW

File: cmd/deploy.go [hunk 1]
```diff
@@ -455,10 +455,13 @@ func deploy(clusterName, version, topoFile string, opt deployOptions) error {
...
 	t := task.NewBuilder().
-		SSHKeyGen(meta.ClusterPath(clusterName, "ssh", "id_rsa")).
-		Parallel(downloadCompTasks...).
-		Parallel(envInitTasks...).
-		Parallel(deployCompTasks...).
+		Step("+ Generate SSH keys",
+			task.NewBuilder().SSHKeyGen(meta.ClusterPath(clusterName, "ssh", "id_rsa")).Build()).
+		ParallelStep("+ Download TiDB components", downloadCompTasks...).
+		Step("+ Initialize target host environments",
+			task.NewBuilder().Parallel(envInitTasks...).Build()).
```
Comment (lonng) [inline cmd/deploy.go:462]:
```suggestion
			task.Parallel(envInitTasks)).
```

File: cmd/deploy.go [hunk 2]
```diff
@@ -455,10 +455,13 @@ func deploy(clusterName, version, topoFile string, opt deployOptions) error {
...
+		Step("+ Copy files",
+			task.NewBuilder().Parallel(deployCompTasks...).Build()).
```
Comment (lonng) [inline cmd/deploy.go:464]: ditto

**Comment by lonng** [comment]
Please add the license header for the new file.

File: cmd/deploy.go (hunk 1 above)
Comment (breeswish) [reply cmd/deploy.go:462]: Looks like it doesn't work

**Pull request merged**